test/unit/finance/bill_variable_for_plan_changed_test.rb

```diff
@@ -3,25 +3,32 @@
 require 'test_helper'
 
 class Finance::BillVariableForPlanChangedTest < ActiveSupport::TestCase
-  attr_reader :contract, :account, :app_plan
+  attr_reader :contract, :app_plan
 
   setup do
-    @contract = FactoryBot.build_stubbed(:application_contract)
-    @account  = FactoryBot.build_stubbed(:simple_account)
+    @contract = FactoryBot.build_stubbed(:contract)
+    account = FactoryBot.build_stubbed(:simple_account)
     @app_plan = FactoryBot.build_stubbed(:application_plan)
 
+    contract.stubs(:provider_account).returns(account)
+    account.stubs(:provider_can_use?).with(:instant_bill_plan_change).returns(true)
+
     @org_tz = ENV["TZ"]
   end
 
   teardown do
     ENV["TZ"] = @org_tz
   end
 
-  test "bill for variable" do
-    Timecop.travel(15.days.ago) if Time.now.mday == 1 # rubocop:disable Rails/TimeZone we don't use timezones in billing
+  test "bill for variable on 1st day of month UTC" do
+    Timecop.travel(Time.now.beginning_of_month) # rubocop:disable Rails/TimeZone we don't use timezones in billing
 
-    contract.stubs(:provider_account).returns(account)
-    account.stubs(:provider_can_use?).returns(true)
+    contract.expects(:save).never
+    contract.notify_observers(:bill_variable_for_plan_changed, app_plan)
+  end
```

Comment:
This one seems to replicate `no variable billing on 1st day of month local time`, with the difference that local timezone on test machine is used instead of a hardcoded one. If you intend to use UTC, then it needs to be set. Default on our test machines is +1 I think.

It should pass either way, just title is suggesting UTC while test runs with whatever timezone happens to be on current machine.

Reply:
Yes I got confused when rebasing. I wanted to remove the whole UTC thing since we don't care about that and I guess I should have removed the whole test case.